llama-index-core/llama_index/core/memory/chat_summary_memory_buffer.py

```diff
@@ -0,0 +1,245 @@
+    def validate_memory(cls, values: dict) -> dict:
+        """Validate the memory."""
+        # Validate token limits
+        token_limit_full_text = values.get("token_limit_full_text", -1)
+        if token_limit_full_text < 1:
+            raise ValueError(
+                "Token limit for full-text messages must be set and greater than 0."
+            )
+
+        # Validate tokenizer -- this avoids errors when loading from json/dict
+        tokenizer_fn = values.get("tokenizer_fn", None)
+        if tokenizer_fn is None:
+            values["tokenizer_fn"] = get_tokenizer()
+
+        return values
+
+    @classmethod
+    def from_defaults(
+        cls,
+        chat_history: Optional[List[ChatMessage]] = None,
+        summarizer_llm: Optional[LLM] = None,
+        chat_store: Optional[BaseChatStore] = None,
+        chat_store_key: str = DEFAULT_CHAT_STORE_KEY,
+        token_limit_full_text: Optional[int] = None,
+        tokenizer_fn: Optional[Callable[[str], List]] = None,
+        summarize_prompt: Optional[str] = None,
+        count_initial_tokens: bool = False,
+    ) -> "ChatSummaryMemoryBuffer":
+        """Create a chat memory buffer from an LLM
+        and an initial list of chat history messages.
+        """
+        if summarizer_llm is not None:
+            context_window = summarizer_llm.metadata.context_window
+            token_limit_full_text = token_limit_full_text or int(
+                context_window * DEFAULT_TOKEN_LIMIT_FULL_TEXT_RATIO
+            )
+        elif token_limit_full_text is None:
+            token_limit_full_text = DEFAULT_TOKEN_LIMIT_FULL_TEXT
+
+        chat_store = chat_store or SimpleChatStore()
+        chat_history = chat_history or []
+        chat_store.set_messages(chat_store_key, chat_history)
+
+        summarize_prompt = summarize_prompt or SUMMARIZE_PROMPT
+        return cls(
+            summarizer_llm=summarizer_llm,
+            token_limit_full_text=token_limit_full_text,
+            # TODO: Check if we can get the tokenizer from the llm
+            tokenizer_fn=tokenizer_fn or get_tokenizer(),
+            summarize_prompt=summarize_prompt,
+            chat_store=chat_store,
+            chat_store_key=chat_store_key,
+            count_initial_tokens=count_initial_tokens,
+        )
+
+    @classmethod
+    def class_name(cls) -> str:
+        """Get class name."""
+        return "ChatSummaryMemoryBuffer"
+
+    def to_string(self) -> str:
+        """Convert memory to string."""
+        return self.json()
+
+    def to_dict(self, **kwargs: Any) -> dict:
+        """Convert memory to dict."""
+        return self.dict()
+
+    @classmethod
+    def from_string(cls, json_str: str) -> "ChatMemoryBuffer":
+        raise NotImplementedError("This is not yet supported.")
+
+    @classmethod
+    def from_dict(
+        cls, data: Dict[str, Any], **kwargs: Any
+    ) -> "ChatSummaryMemoryBuffer":
+        raise NotImplementedError("This is not yet supported.")

```

Comment:
from_dict() / from_string() could be implemented by just passing in the LLM when calling it?

Reply:
I'm not sure if I understood you correctly, but I added a basic implementation. If this is not what you are looking for, could you make the code change?